src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -470,23 +624,26 @@ private void UpdatePackageList(IEnumerable<PackageItemListViewModel> packages, b
                 {
                     // remove the loading status indicator if it's in the list
                     bool removed = Items.Remove(_loadingStatusIndicator);
-
+                    System.Diagnostics.Debug.WriteLine("UpdatePackageList001 was (" + Items.Count + ")"); //TODO remove
                     if (refresh)
                     {
                         ClearPackageList();
                     }
+                    System.Diagnostics.Debug.WriteLine("UpdatePackageList002 Cleared (" + Items.Count + ")"); //TODO remove
 
                     // add newly loaded items
                     foreach (var package in packages)
                     {
                         package.PropertyChanged += Package_PropertyChanged;
                         Items.Add(package);
```

Comment:
@nkolev92 the biggest pointer I think I have right now is that this adding of packages is fine... but sometimes the package.IsUpdateAvailable is false when it should be true. The reason is it's a calculated property based on package.Status. Sometimes that's not loaded yet. I think b/c my UI filtering is happening before loading is done?